tests/fermi/test_fermionic.py

```diff
@@ -96,37 +97,268 @@ def test_compact(self, fw, str_rep):
+        """Test that a FermiWord can be multiplied onto a number (from the right)
+        and return a FermiSentence"""
+        assert number * fw == result
 
     tup_fw_mult_error = (
         (fw1, [1.5]),
         (fw4, "string"),
     )
 
-    @pytest.mark.parametrize("f1, f2", tup_fw_mult_error)
-    def test_mul_error(self, f1, f2):
-        with pytest.raises(TypeError, match=f"Cannot multiply FermiWord by {type(f2)}."):
-            f1 * f2  # pylint: disable=pointless-statement
+    @pytest.mark.parametrize("fw, bad_type", tup_fw_mult_error)
+    def test_mul_error(self, fw, bad_type):
+        """Test multiply with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot multiply FermiWord by {type(bad_type)}."):
+            fw * bad_type  # pylint: disable=pointless-statement
+
+    @pytest.mark.parametrize("fw, bad_type", tup_fw_mult_error)
+    def test_rmul_error(self, fw, bad_type):
+        """Test __rmul__ with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot multiply FermiWord by {type(bad_type)}."):
+            bad_type * fw  # pylint: disable=pointless-statement
+
+    @pytest.mark.parametrize("fw, bad_type", tup_fw_mult_error)
+    def test_add_error(self, fw, bad_type):
+        """Test __add__ with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot add {type(bad_type)} to a FermiWord"):
+            fw + bad_type  # pylint: disable=pointless-statement
+
+    @pytest.mark.parametrize("fw, bad_type", tup_fw_mult_error)
+    def test_sub_error(self, fw, bad_type):
+        """Test __sub__ with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot subtract {type(bad_type)} from a FermiWord"):
+            fw - bad_type  # pylint: disable=pointless-statement
+
+    WORDS_ADD = [
+        (fw1, fw2, FermiSentence({fw1: 1, fw2: 1})),
+        (fw3, fw2, FermiSentence({fw2: 1, fw3: 1})),
+        (fw2, fw2, FermiSentence({fw2: 2})),
+    ]
+
+    @pytest.mark.parametrize("f1, f2, res", WORDS_ADD)
+    def test_add_fermi_words(self, f1, f2, res):
+        """Test that adding two FermiWords returns the expected FermiSentence"""
+        assert f1 + f2 == res
+        assert f2 + f1 == res
+
+    WORDS_AND_SENTENCES_ADD = [
+        (fw1, FermiSentence({fw1: 1.2, fw3: 3j}), FermiSentence({fw1: 2.2, fw3: 3j})),
+        (fw3, FermiSentence({fw1: 1.2, fw3: 3j}), FermiSentence({fw1: 1.2, fw3: (1 + 3j)})),
+        (fw1, FermiSentence({fw1: -1.2, fw3: 3j}), FermiSentence({fw1: -0.2, fw3: 3j})),
+    ]
+
+    @pytest.mark.parametrize("w, s, res", WORDS_AND_SENTENCES_ADD)
+    def test_add_fermi_words_and_sentences(self, w, s, res):
+        """Test that adding a FermiSentence to a FermiWord returns the expected FermiSentence"""
+        sum = w + s
+        # due to rounding, the actual result for floats is
+        # e.g. -0.19999999999999... instead of 0.2, so we round to compare
```

Comment:
There might be a better way? Maybe using more digits after the decimal point, e.g., `1.231` instead of `1.2`?

Reply:
I can try that and see :)

Reply:
Then it tries to compare `-0.2310000000...` with -0.231 and says they aren't equal 😢